vm/src/function/number.rs

```diff
@@ -125,3 +126,32 @@ impl TryFromObject for ArgIntoBool {
         })
     }
 }
+
+// Implement ArgIndex to seperate between "true" int and int generated by index
+#[derive(Debug, PartialEq)]
+#[repr(transparent)]
+pub struct ArgIndex {
+    value: BigInt,
+}
+
+impl From<ArgIndex> for BigInt {
+    fn from(arg: ArgIndex) -> Self {
+        arg.value
+    }
+}
+
+impl Deref for ArgIndex {
+    type Target = BigInt;
+
+    fn deref(&self) -> &Self::Target {
+        &self.value
+    }
+}
+
+impl TryFromObject for ArgIndex {
+    fn try_from_object(vm: &VirtualMachine, obj: PyObjectRef) -> PyResult<Self> {
+        Ok(Self {
+            value: obj.try_index(vm)?.as_bigint().clone(),
+        })
+    }
+}
```

Comment:
...
using `TryFromBorrowedObject`. Either way we can avoid an additional clone.
```rust
impl TryFromBorrowedObject for ArgIndex {
    ...
```